
# **タイタニック号生存者の予測**
「ランダムフォレスト」という機械学習のアルゴリズムにより、タイタニック号の生存者を予測する。

In [ ]:
import numpy as np
import pandas as pd

ｄｆ_train = pd.read_csv('train.csv')
ｄｆ_test = pd.read_csv('test.csv')

ｄｆ_train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
ｄｆ_test.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

# **データの前処理**

In [ ]:
test_id = df_test['PassengerId']

ｄｆ_train_data = ｄｆ_train.drop(['PassengerId', 'Name', 'Ticket', 'Cabin'], axis=1)
ｄｆ_test_data = ｄｆ_test.drop(['PassengerId', 'Name', 'Ticket', 'Cabin'], axis=1)

In [ ]:
ｄｆ_train_data

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0           0       3    male  22.0      1      0   7.2500        S
1           1       1  female  38.0      1      0  71.2833        C
2           1       3  female  26.0      0      0   7.9250        S
3           1       1  female  35.0      1      0  53.1000        S
4           0       3    male  35.0      0      0   8.0500        S
..        ...     ...     ...   ...    ...    ...      ...      ...
886         0       2    male  27.0      0      0  13.0000        S
887         1       1  female  19.0      0      0  30.0000        S
888         0       3  female   NaN      1      2  23.4500        S
889         1       1    male  26.0      0      0  30.0000        C
890         0       3    male  32.0      0      0   7.7500        Q

[891 rows x 8 columns]

In [ ]:
ｄｆ_test_data

Pclass     Sex   Age  SibSp  Parch      Fare Embarked
0         3    male  34.5      0      0    7.8292        Q
1         3  female  47.0      1      0    7.0000        S
2         2    male  62.0      0      0    9.6875        Q
3         3    male  27.0      0      0    8.6625        S
4         3  female  22.0      1      1   12.2875        S
..      ...     ...   ...    ...    ...       ...      ...
413       3    male   NaN      0      0    8.0500        S
414       1  female  39.0      0      0  108.9000        C
415       3    male  38.5      0      0    7.2500        S
416       3    male   NaN      0      0    8.0500        S
417       3    male   NaN      1      1   22.3583        C

[418 rows x 7 columns]

In [ ]:
ｄｆ_train_data.info()
ｄｆ_test_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Sex       891 non-null    object 
 3   Age       714 non-null    float64
 4   SibSp     891 non-null    int64  
 5   Parch     891 non-null    int64  
 6   Fare      891 non-null    float64
 7   Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(2)
memory usage: 55.8+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Pclass    418 non-null    int64  
 1   Sex       418 non-null    object 
 2   Age       332 non-null    float64
 3   SibSp     418 non-null    int64  
 4   Parch     418 non-null    int64  
 5   Fare      417 non-null    float64
 6   Embarked  418 non-null

# **欠損値**

df_train_data
*   3   Age       714 non-null    float64
*   7   Embarked  889 non-null    object 

df_test_data

*   2   Age       332 non-null    float64
*   5   Fare      417 non-null    float64


In [ ]:
# Age
age_mean = df_train_data["Age"].mean()  # 平均値
df_train_data["Age"] = df_train_data["Age"].fillna(age_mean)
df_test_data["Age"] = df_test_data["Age"].fillna(age_mean)

# Fare
fare_mean = df_train_data["Fare"].mean()  # 平均値
df_train_data["Fare"] = df_train_data["Fare"].fillna(fare_mean)
df_test_data["Fare"] = df_test_data["Fare"].fillna(fare_mean)

# Embarked
embarked_mode = df_train_data["Embarked"].mode()  # 最頻値
df_train_data["Embarked"] = df_train_data["Embarked"].fillna(embarked_mode)
df_test_data["Embarked"] = df_test_data["Embarked"].fillna(embarked_mode)

# **ダミー変数**
文字列などのカテゴリ変数に対しては、数値に変換する

In [ ]:
change_labels = ["Sex", "Embarked"]  # カテゴリ変数のラベル
train_data = pd.get_dummies(df_train_data, columns=change_labels)
test_data = pd.get_dummies(df_test_data, columns=change_labels)

train_data.head()

Survived  Pclass   Age  SibSp  Parch     Fare  Sex_female  Sex_male  \
0         0       3  22.0      1      0   7.2500           0         1   
1         1       1  38.0      1      0  71.2833           1         0   
2         1       3  26.0      0      0   7.9250           1         0   
3         1       1  35.0      1      0  53.1000           1         0   
4         0       3  35.0      0      0   8.0500           0         1   

   Embarked_C  Embarked_Q  Embarked_S  
0           0           0           1  
1           1           0           0  
2           0           0           1  
3           0           0           1  
4           0           0           1

In [ ]:
t_train = train_data["Survived"]  # 正解
X_train = train_data.drop(labels=["Survived"], axis=1)  # "Survived"の列を削除して入力に
X_test = test_data

# **学習フェーズ**

In [ ]:
from sklearn.svm import SVC

model = SVC()

model.fit(X_train, t_train)

SVC()

In [ ]:
y_test = model.predict(X_test)

# 形式を整える
survived_test = pd.Series(y_test, name="Survived")
subm_data = pd.concat([test_id, survived_test], axis=1)

# 提出用のcsvファイルを保存
subm_data.to_csv("submission_titanic.csv", index=False)

subm_data

PassengerId  Survived
0            892         0
1            893         0
2            894         0
3            895         0
4            896         0
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

In [ ]:
ACCESS_TOKEN = "コピーしたトークンをここに貼り付け"
REPOSITORY = "smiyawaki0820/sample"  # clone/push したいリポジトリ
USER = "smiyawaki0820"
MAIL = "lab.skow.mywk@gmail.com"

WORKDIR = "sample"
BRANCH = "develop"

In [ ]:
! git clone https://{USER}:{ACCESS_TOKEN}@github.com/{REPOSITORY}.git {WORKDIR}
%cd {WORKDIR}
! git config --global user.name {USER}
! git config --global user.email {MAIL}
! git remote set-url origin https:/{USER}:{ACCESS_TOKEN}@github.com/{REPOSITORY}.git
! git checkout -b {BRANCH}
! git branch

In [ ]:
%cd /content/gdrive/My Drive/#add your custom working directory
!pwd
!git init
!git remote add origin https://{github_username}:{github_password}@github.com/{github_username}/{github_repo_name}.git
!git remote -v
!git config --global user.email '{user_email}'
!git config --global user.name '{user_name}'
!git add .
!git commit -m 'First commit'
!git push origin master